### DISCUSSION TITLE
To enable GitHub Copilot, authorize this extension using GitHub's device flow

### DISCUSSION THREAD

**Discussion by ghost** [community/community#65913] [Copilot Conversations]
### Select Topic Area

Question

### Body

When I open visual studio I get this error message and a button "Sign in with Device Flow" which takes me to this page:
https://github.com/login/device

...where I am asked to "Enter the code displayed on your device"

What device?

I have not been asked to set up any device and there is no link or information on how to set it up.

![github](https://github.com/community/community/assets/129844120/f64fdf79-29a4-4ffe-babe-438437191941)

**Comment by arafajs** [comment]
Check your Email.

**Reply by ghost** [reply]
Not received any email. I have tried a couple of times with a few days between.

**Reply by MalxWare47** [reply]
i HAVE THE SAME SITUATION WHEN I JUST UPDATED VISUAL STUDIO AND WHAT THE HELL IS DEVICE FLOW AND WHAT CODE WTFFF

**Reply by JasonTan94** [reply]
It's in your clipboard, just Ctrl + V will do. Or you can find the code in your output terminal

**Reply by ghost** [reply]
I had to reinstall my PC for another reason, and after installing Visual Studio again, I was not asked to use Device Flow.

**Reply by frodoeric** [reply]
The device activation code is in the Output Visual Studio console 
![image](https://github.com/community/community/assets/11093562/4fc32009-7eb3-4b2a-8a19-7a2134150614)

**Reply by xlisses** [reply]
THX bro

**Comment by UnusualJustin** [comment]
This is an absolutely horrible user experience. I spent 2 days trying to get copilot to work after a visual studio update. I was on the verge of cancelling my subscription.

**Comment by jesualujan** [comment]
i'm trying to activate copilot cli on hyper terminal:
Then go to https://github.com/login/device, paste the code in and approve the access.
Failed to authenticate: getaddrinfo ENOTFOUND next-waitlist.azurewebsites.net, undefined

**Reply by Baojch** [reply]
same question, any solutions?

**Reply by leonhartyao** [reply]
Same issue here

**Comment by Johnny1110** [comment]
That one was the most stupid feature that I ever seen.

**Reply by Anthonioxz** [reply]
grok

**Reply by Anthonioxz** [reply]
[]()

**Reply by Anthonioxz** [reply]
[]()
